classes/local/store/object_file_system.php

```diff
@@ -339,19 +341,39 @@ public function delete_empty_dirs($rootpath = '') {
         }
         $empty = true;
         foreach (glob($rootpath . DIRECTORY_SEPARATOR . '*') as $path) {
-            // If there are .tmp files here, they should be killed.
-            if (!is_dir($path) && pathinfo($path, PATHINFO_EXTENSION) === 'tmp') {
+            if (!is_dir($path)) {
                 // Check timemodified, don't touch anything more recent than 24 hours.
                 $modified = filemtime($path);
                 if ($modified === false) {
                     $modified = 0;
                 }
-                $delete = $modified <= (time() - DAYSECS);
 
-                if ($delete) {
-                    @unlink($path);
+                if ($modified <= (time() - DAYSECS)) {
+                    $pathinfo = pathinfo($path);
+
+                    // If there are .tmp files here, they should be killed.
+                    if (!empty($pathinfo['extension']) && $pathinfo['extension'] === 'tmp') {
+                        @unlink($path);
+                    } else {
+                        // If the file is 24hrs old, they may not be tracked by objectFS.
+                        // This means it  may not exist in the files table at all.
```

Comment:
double space in comment

Also wondering about performance if you have chosen to set the sitethreshold setting, we will constantly keep doing db calls grinding through them. Not sure best approach here

Reply:
To combat this, I have avoided any files that are underneath the size threshold. It will still do a little bit of comparison on them, but nowhere near as heavy as a db call to check them out (plus it only runs once a week). You get very little benefit from the tiny files anyway, unless there are millions of them.

This issue also appears if deletelocal is disabled, as the filedir will contain massive amounts of files. In this instance, I've opted just to skip the checks, as the likely improvement is only a few %, and it will take a massive amount of time to process.